### DISCUSSION TITLE
Example of Combobox with API-driven matches?

### DISCUSSION THREAD

**Discussion by agriffis** [ariakit/ariakit#4214] [Questions]
I'm working through an app, replacing existing components with AriaKit for the sake of accessibility and easier feature development.

One of the components is a search input that offers suggestions:

https://github.com/user-attachments/assets/42e040f8-9a64-4d10-a288-921e8f540a17

I think the AriaKit component for this would be Combobox, right? Are there any examples of API-driven match generation currently?

**Comment by woble** [comment]
https://ariakit.org/examples/combobox-filtering
https://ariakit.org/examples/combobox-filtering-integrated (you need Plus to see the example)

You can also you use this to do highlight of typed value in an item. Or you can implement your own highlighting.
https://ariakit.org/reference/combobox-item-value

**Reply by agriffis** [reply]
Thanks! My question wasn't very clear. I've seen those examples. Neither is calling an API to generate the list of matches, and I was wondering if there are any examples in Ariakit that do so.

It's no problem, though, I'll figure it out. It's just so convenient that Ariakit has such a wonderful library of examples as starting points! :smile:

**Reply by diegohaz** [reply]
@agriffis We need to build that example. In short, it doesn't matter how you obtain the data. Ariakit is only concerned with the `<ComboboxItem>` elements you render. In other words, you can use any method to fetch the data as long as you output them using `<ComboboxItem>`.

Here's an example using React Query: https://stackblitz.com/edit/n3g3pb?file=combobox-filtering%2Fstyle.css,combobox-filtering%2Findex.tsx&theme=dark
